### PR TITLE
pythonPackages.aiometer

### PR DESCRIPTION
**Pull request by gwg313**
## Things done

- Built on platform(s)
  - [x ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by nagy — CHANGES_REQUESTED** [review]
The commit message needs to follow the guidelines at https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions

File: pkgs/development/python-modules/aiometer/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    owner = "florimondmanca";
+    repo = "aiometer";
+    rev = version;
+    hash = "sha256-Db6AWNWZ5AhMFrbjBRVOFlDpzegsqNL3k5dmTHrjSi8=";
+  };
+
+  nativeBuildInputs = [
+    python3.pkgs.setuptools
+    python3.pkgs.setuptools-scm
+    python3.pkgs.wheel
```
Comment (pbsds) [inline pkgs/development/python-modules/aiometer/default.nix:21]:
```suggestion
    setuptools
    setuptools-scm
    wheel
```

File: pkgs/development/python-modules/aiometer/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
+{ lib
+, python3
+, fetchFromGitHub
+}:
```
Comment (pbsds) [inline pkgs/development/python-modules/aiometer/default.nix:4]:
```suggestion
, pytestCheckHook
, setuptools
, setuptools-scm
, wheel
, anyio
, typing-extensions
}:
```

File: pkgs/development/python-modules/aiometer/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    hash = "sha256-Db6AWNWZ5AhMFrbjBRVOFlDpzegsqNL3k5dmTHrjSi8=";
+  };
+
+  nativeBuildInputs = [
+    python3.pkgs.setuptools
+    python3.pkgs.setuptools-scm
+    python3.pkgs.wheel
+  ];
+
+  propagatedBuildInputs = with python3.pkgs; [
```
Comment (pbsds) [inline pkgs/development/python-modules/aiometer/default.nix:24]:
```suggestion
  propagatedBuildInputs = [
```

File: pkgs/development/python-modules/aiometer/default.nix [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    anyio
+    typing-extensions
+  ];
+
```
Comment (pbsds) [inline pkgs/development/python-modules/aiometer/default.nix:28]:
```suggestion

  nativeCheckInputs = [
    pytestCheckHook
  ];

```

File: pkgs/development/python-modules/aiometer/default.nix [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  version = "0.4.0";
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "florimondmanca";
+    repo = "aiometer";
+    rev = version;
+    hash = "sha256-Db6AWNWZ5AhMFrbjBRVOFlDpzegsqNL3k5dmTHrjSi8=";
+  };
+
```
Comment (pbsds) [inline pkgs/development/python-modules/aiometer/default.nix:17]: Something like this is likely needed:

```suggestion

  # disable coverage testing
  postPatch = ''
    substituteInPlace setup.cfg \
      --replace "--cov" "#--cov"
  '';

```

File: pkgs/development/python-modules/aiometer/default.nix [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  ];
+
+  pythonImportsCheck = [ "aiometer" ];
+
+  meta = with lib; {
+    description = "A Python concurrency scheduling library, compatible with asyncio and trio";
+    homepage = "https://github.com/florimondmanca/aiometer";
+    changelog = "https://github.com/florimondmanca/aiometer/blob/${src.rev}/CHANGELOG.md";
+    license = licenses.mit;
+    maintainers = with maintainers; [ gwg313 ];
```
Comment (pbsds) [inline pkgs/development/python-modules/aiometer/default.nix:36]: Seems like `gwg313` is not defined in `maintainers/maintainer-list.nix`. Please add yourself in a separate commit titled `maintainers: add gwg313`, preceeding the aiometer commit.
See https://github.com/NixOS/nixpkgs/tree/master/maintainers#individual-maintainer-list

**Review by pbsds — CHANGES_REQUESTED** [review]
Looking good. Welcome to nixpkgs.

Please reword the commit and PR title to `python3Packages.aiometer: init at 0.4.0`.

**Pull request closed without merging**